code/modules/surgery/organs/tongue.dm

```diff
@@ -318,3 +318,46 @@
 /obj/item/organ/tongue/ethereal/Initialize(mapload)
 	. = ..()
 	languages_possible = languages_possible_ethereal
+
+//Sign Language Tongue - yep, that's how you speak sign language.
+/obj/item/organ/tongue/tied
+	name = "tied tongue"
+	desc = "If only one had a sword so we may finally untie this knot."
+	say_mod = "signs"
+	icon_state = "tonguetied"
+	modifies_speech = TRUE
+
+/obj/item/organ/tongue/tied/Insert(mob/living/carbon/M)
+	. = ..()
+	M.verb_ask = "signs"
+	M.verb_exclaim = "signs"
+	M.verb_whisper = "subtly signs"
+	M.verb_sing = "rythmically signs"
+	M.verb_yell = "emphatically signs"
+	ADD_TRAIT(M, TRAIT_SIGN_LANG, "tongue")
+	REMOVE_TRAIT(M, TRAIT_MUTE, "tongue")
+
+/obj/item/organ/tongue/tied/Remove(mob/living/carbon/M, special = 0)
+	..()
+	REMOVE_TRAIT(M, TRAIT_SIGN_LANG, "tongue") //People who are Ahealed get "cured" of their sign language-having ways. If I knew how to make the tied tongue persist through aheals, I'd do that.
+
+//Thank you Jwapplephobia for helping me with the literal hellcode below
+/obj/item/organ/tongue/tied/handle_speech(datum/source, list/speech_args)
+	var/new_message
+	var/message = speech_args[SPEECH_MESSAGE]
+	var/exclamation_used = FALSE
+	var/question_mark_used = FALSE
+	var/mob/living/carbon/M = owner
+	for(var/i in 1 to length(message))
+		if(findtext("!", message[i]))
+			exclamation_used = TRUE
+		else if (findtext("?", message[i]))
+			question_mark_used = TRUE
+		else
+			new_message += message[i]
+	if(exclamation_used)
+		M.visible_message("<span class='notice'>[M] raises [M.p_their()] eyebrows.</span>")
+	if(question_mark_used)
+		M.visible_message("<span class='notice'>[M] lowers [M.p_their()] eyebrows.</span>")
+	speech_args[SPEECH_MESSAGE] = new_message
```

Comment:
Aight here we gooooooo.
Iterating through each char will get costly fast, let's do this instead.
`findtext()` can check the whole string, so we don't need to worry about that anymore.
`Replace()`(Regex) or `replacetext()` can replace all instances of something in a string, if you set it to replace with "" it'll just remove it.

You can put the findtexts in the if checks in place of exclamation_used, and set speech_args[SPEECH_MESSAGE] to the replaced text.

Here are the links to the documentation for these procs 
http://www.byond.com/docs/ref/#/proc/findtext
http://www.byond.com/docs/ref/#/proc/replacetext
http://www.byond.com/docs/ref/info.html#/regex/proc/Replace

Even tho they do internal iteration, I belive they'll be faster, just because they're internal procs, and it'll help clean up your code a good deal.

I really ought to profile this, but I can't get my local to profile properly, this shouldn't be a horribly hot proc in any case, but if you want to test it out feel free.

Reply:
We thought about this, but the one thing we're more worried over if we change it is if we'll still be able to get an output for `question_mark_used` & `exclamation_used` so we can put those messages into chat rather than the punctuation. If the text is replaced using `replacetext()`, we wouldn't be able to get those `if` statements, right?

Reply:
before you use removetext() to get rid of the !/?'s, couldn't you do something like
```
if(findtext(message, "!")
thing
if(findtext(message, "?")
other thing
```
?

Reply:
Thoughts?

Reply:
Oh, I fixed that a while ago! Sorry I didn't say anything. I believe the new code is here?